hedera-node/configuration/compose/settings.txt

```diff
@@ -12,4 +12,4 @@ state.saveStatePeriod,                         300
 useLoopbackIp,                                 false
 waitAtStartup,                                 false
 virtualMap.preferredFlushQueueSize,            10000
-state.mainClassNameOverride,                   com.hedera.services.ServicesMain
\ No newline at end of file
+state.mainClassNameOverride,                   com.hedera.node.app.ServicesMain
```

Comment:
I think this should not be changed. Because Devops uses this path to use their old scripts.

Reply:
Perhaps an odd question to add to this.  Why are we using a class name in a path (or data state, for that matter)?  Wouldn't it be better to have the path be a fixed (perhaps configurable) value that has no connection to the name of a class?

Reply:
I agree this seems "sus" as the kids say :-). I think I know the history of it though.

The original "Swirlds" platform concept was a browser concept. There would be a browser (not Web browser, but a Swirlds browser) which given a reference to a Swirld (maybe a URL?) it would load the jars dynamically and run the Swirld within the Browser. A single Browser could have multiple Swirlds (swirld == shared world). And for logs and files related to the swirld, it would get written to disk. Using the package name by convention would allow multiple Swirlds in one Browser to save files to different places.

A more robust implementation would need another way to do this, because two swirlds could use the same package / main class name and collide. But I think that's where it started and then there are hardcoded paths in places that refer to that path and changing those paths blows things up. That's my guess.

Reply:
Currently, we have this overridden our configs in places such as https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/configuration/preprod/settings.txt#L30 & https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/configuration/mainnet/settings.txt#L29

If that continues to be safe behavior after this PR, then I think we're OK. We can start working towards updating our automation with the goal of adopting the default value within the next couple of releases.

If this will necessitate an immediate change in settings, then I think we should have a conversation between release engineering, devops, and eng to determine all the touch points and coordinate this change since I imagine a lot of places may be affected between DevOps tooling, JRS, NMT, etc., and any one of them could cause problems.

Reply:
agreed with @beeradb , any change to state save path needs to be planned work for several teams

as for this particular file change, does the compose deployment have anything that relies on the current state save path? that would be the only concern here I believe. this particular file is not used for any preprod/prod environments

Reply:
Should I just revert all changes to settings.txt files? If they don't have to match the new package name, I can put them back

Reply:
OK I have reverted these back